### DISCUSSION TITLE
Predictions on unseen Data

### DISCUSSION THREAD

**Discussion by PyRSQL** [DoubleML/doubleml-for-py#252] [Q&A] [answered]
Hello DoubleML-Team,

i am currently working on my bachelor thesis and need to make predictions using the `doubleml` library in Python. My objective is to validate my price elasticities through a train-test split approach. Specifically, I plan to use 2.5 years of data for training with the doubleml 
library and then test the model on the subsequent 0.5 years of unseen data. This will help me evaluate the accuracy of the price elasticity predictions.

Is it possible to implement this methodology using the doubleml library?

Best Regards,
David

**Comment by JanTeichertKluge** [comment, marked as answer]
Hi David,

Thank you for your question. The latest version of `DoubleML` offers (for many models) the possibility to pass external predictions for the nuisance functions. So you could use a classic train-test split instead of the cross-fitting method.
You train the ML models on your training dataset and have the predictions for the nuisance functions of the test dataset generated. 
These in turn can then be passed to the `DoubleMLPLR` model which then estimates the causal parameter $\theta$, for example.
Further information can be found here: https://docs.doubleml.org/stable/guide/learners.html#advanced-external-predictions
I hope I was able to answer your question.

Thanks and best regards
Jan
